website/src/components/McapRecordingDemo/videoCapture.ts

```diff
@@ -66,40 +85,189 @@ export function startVideoCapture(params: VideoCaptureParams): () => void {
   };
 }
 
+/**
+ * Sometimes `VideoEncoder.isConfigSupported` returns true but the encoder does not actually output
+ * frames (looking at you, Safari). This function tries actually encoding a frame and making sure
+ * that the encoder can output a chunk.
+ */
+async function isEncoderConfigActuallySupported(config: VideoEncoderConfig) {
+  try {
+    if ((await VideoEncoder.isConfigSupported(config)).supported !== true) {
+      return false;
+    }
+    let outputAnyFrames = false as boolean;
+    let hadErrors = false as boolean;
+    const encoder = new VideoEncoder({
+      output() {
+        outputAnyFrames = true;
+      },
+      error(err) {
+        hadErrors = true;
+        console.log(
+          "VideoEncoder error during compatibility detection:",
+          config,
+          err,
+        );
+      },
+    });
+    encoder.configure(config);
+    const bitmap = await createImageBitmap(new ImageData(1, 1));
+    const frame = new VideoFrame(bitmap, { timestamp: 0 });
+    bitmap.close();
+    encoder.encode(frame, { keyFrame: true });
```

Comment:
i guess it's simplest to encode all frames as keyframes, but if someone's using this demo to compare CompressedImage vs. VideoFrame, won't they see them as exactly the same?